virttest/qemu_vm.py

```diff
@@ -1467,23 +1488,13 @@ def sort_key(dev):
 
         vga = params.get("vga")
         if vga:
-            if vga != 'none':
-                devices.insert(StrDev('VGA-%s' % vga,
-                                      cmdline=add_vga(vga),
-                                      parent_bus={'aobject': 'pci.0'}))
-                if vga == 'qxl':
-                    qxl_dev_nr = int(params.get("qxl_dev_nr", 1))
-                    if qxl_dev_nr > 1:
-                        addr = int(params.get("qxl_base_addr", 29))
-                        cmdline = add_qxl(qxl_dev_nr, addr)
-                        devices.insert(StrDev('qxl', cmdline=cmdline))
-            else:
-                devices.insert(StrDev('VGA-none', cmdline=add_vga(vga)))
-
-        elif params.get('defaults', 'no') != 'no':  # by default add cirrus
-            devices.insert(StrDev('VGA-cirrus',
-                                  cmdline=add_vga(vga),
-                                  parent_bus={'aobject': 'pci.0'}))
+            devices.insert(add_vga(devices, vga))
+            if vga == 'qxl':
```

Comment:
just a minor comment, now that you have handled most of the logic inside add_vga, can't this also handled inside the same function?

Reply:
this `if` clause actually intends to add the secondary `qxl` device when using `qxl` as the primary vga device, yes it indeed needs an improvement since now we have several type of secondary vga devices, but it is another issue and I'd suggest to handle it separately (the basic idea is to change `add_qxl()` to `add_secondary_vga()`).

Reply:
@luckyh Agree, we can handle as separate PR, LGTM otherwise.